tests/imports.nim

```diff
@@ -0,0 +1,46 @@
+# Nim-LibP2P
+# Copyright (c) 2023-2025 Status Research & Development GmbH
+# Licensed under either of
+#  * Apache License, version 2.0 ([LICENSE-APACHE](LICENSE-APACHE))
+#  * MIT license ([LICENSE-MIT](LICENSE-MIT))
+# at your option.
+# This file may not be copied, modified, or distributed except according to
+# those terms.
+
+import std/macros
+import std/os
+import std/strutils
+import std/sequtils
+
+proc buildImports(
+    dir: string, ignorePaths: seq[string] = @[], matchPath: string = ""
+): NimNode =
+  let imports = newStmtList()
+  for file in walkDirRec(dir):
+    let (path, name, ext) = splitFile(file)
+    let isTestFile = name.startsWith("test_") and not name.startsWith("test_all")
+    let isNimFile = ext == ".nim"
+    let isIgnored = ignorePaths.len > 0 and ignorePaths.anyIt(path.contains(it))
+    let isMatched = matchPath.len == 0 or file.contains(matchPath)
+
+    if isTestFile and isNimFile and not isIgnored and isMatched:
+      imports.add(newNimNode(nnkImportStmt).add(newLit(file)))
+  imports
+
+macro importTests*(dir: static string): untyped =
+  ## imports all files in the specified directory whose filename
+  ## starts with "test_" and ends in ".nim"
+  ## ignores test_all files
+  buildImports(dir, @[])
+
+macro importTests*(dir: static string, ignorePaths: static seq[string]): untyped =
+  ## imports all files in the specified directory whose filename
+  ## starts with "test_" and ends in ".nim"
+  ## ignores test_all files and any paths listed in ignorePaths
+  buildImports(dir, ignorePaths)
+
+macro importTestsMatching*(dir: static string, singlePath: static string): untyped =
+  ## imports only files in the specified directory whose path contains singlePath
+  ## and whose filename starts with "test_" and ends in ".nim"
+  ## ignores test_all files
+  buildImports(dir, @[], singlePath)
```

Comment:
why have 2 overloaded macros and one macro with different name? why can't it be single macro or just call proc directly ?